crates/task-impls/src/consensus.rs

```diff
@@ -609,11 +610,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                             // starting from the first iteration with a three chain, e.g. right after the else if case nested in the if case above
                             if new_decide_reached {
                                 let mut leaf = leaf.clone();
-                                consensus
-                                    .metrics
-                                    .last_synced_block_height
-                                    .set(usize::try_from(leaf.get_height()).unwrap_or(0));
-
+                                if !is_ancestor {
```

Comment:
I think this makes more sense. But if I update metrics at line 677 the synced block height will be much higher than updating at line 613, why this could be the case?

Reply:
Oh, my understanding is that the `leaf` we called on 677 is the current newest leaf, we made a decide on this view = leaf.view(), but the decided block is the leaf on view - 2 (in most cases) due to the consensus protocol. Therefore we'd better still do the update on line 613. Is this understanding correct?